test/lib/functionTester.js

```diff
@@ -0,0 +1,35 @@
+module.exports = createTestFunction;
+
+function createTestFunction( shouldBeCalled, times ){
+  if( times > 0 && !shouldBeCalled )
+    console.log( "WARNING : Trying to test if it should not be called with times > 0",
+                 ( new Error() ).stack );
+
+  var numberOfCalls = 0;
+
+  var f = function testFunction(){
```

Comment:
Is there a reason to store the function in a variable instead of just declaring it?

```js
function testFunction() {

}

testFunction.hasBeenCalled = function() {
}

testFunction.numberOfCalls = function() {
}

testFunction.testFunctionCall = function() {
}

```